src/rules/import-alias.ts

```diff
@@ -10,23 +10,59 @@ import type { JSONSchema4 } from "json-schema";
 import { dirname, join as joinPath, resolve, sep as pathSep } from "path";
 import slash from "slash";
 
+const isPathImport = (
+    relativeImportOverride: RelativeImportConfig
+): relativeImportOverride is RelativePathConfig => {
+    return (relativeImportOverride as RelativePathConfig).path !== undefined;
+};
+
+const isGlobImport = (
+    relativeImportOverride: RelativeImportConfig
+): relativeImportOverride is RelativeGlobConfig => {
+    return (relativeImportOverride as RelativeGlobConfig).pattern !== undefined;
+};
+
 function isPermittedRelativeImport(
     importModuleName: string,
     relativeImportOverrides: RelativeImportConfig[],
-    fileAbsoluteDir: string
+    fileAbsoluteDir: string,
+    filepath: string
 ) {
-    const configOfPath = relativeImportOverrides.find((config) =>
-        fileAbsoluteDir.includes(resolve(config.path))
-    );
-    if (!configOfPath) {
-        return false;
-    }
+    console.log({
+        importModuleName,
+        relativeImportOverrides,
+        fileAbsoluteDir,
+        filepath,
+    });
+    for (const relativeImportOverride of relativeImportOverrides) {
+        const importParts = importModuleName.split("/");
+        const relativeDepth = importParts.filter(
+            (moduleNamePart) => moduleNamePart === ".."
+        ).length;
+
+        if (isPathImport(relativeImportOverride)) {
+            const configOfPath = fileAbsoluteDir.includes(
+                resolve(relativeImportOverride.path)
+            );
+            if (!configOfPath) {
+                return false;
+            }
```

Comment:
It looks like this shouldn't work in the case where there are multiple paths/glob configurations. The current loop would always return true/false on the first item of the array instead of checking through them all. You might want a `continue`; although, I might suggest to simply update the closure inside of `find` from the current commit's code.

Reply:
While that sounds reasonable, I'm a little concerned we might try to be too clever. For simplicity's sake, could we simply prefer the highest configured `depth` that matches? That way it'll be simply on the user to define their configuration in the way they want it.